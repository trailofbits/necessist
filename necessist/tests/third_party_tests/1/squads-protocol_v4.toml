url = "https://github.com/Squads-Protocol/v4"

target_os = ["linux", "macos"]

# smoelius: `yarn build && yarn pretest` must be run before `anchor test`.
init = "yarn && yarn build && yarn pretest"

framework = "anchor"

source_files = [
    "tests/suites/examples/create-mint.ts",
    "tests/suites/examples/immediate-execution.ts",
]

full = true

args = [
    "--allow=instrumentation-nonbuildable",
    "--",
    "--",
    "--features=testing",
]
