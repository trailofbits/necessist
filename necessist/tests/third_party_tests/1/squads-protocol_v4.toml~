url = "https://github.com/Squads-Protocol/v4"

rev = "19597dc082986d7d95a89ee124dc23625b1b0039"

target_os = ["linux", "macos"]

# smoelius: `yarn build && yarn pretest` must be run before `anchor test`.
init = "yarn && yarn build && yarn pretest"

framework = "anchor"

source_files = [
    "tests/suites/examples/create-mint.ts",
    "tests/suites/examples/immediate-execution.ts",
]

args = [
    "--allow=instrumentation-nonbuildable",
    "--",
    "--",
    "--features=testing",
]
