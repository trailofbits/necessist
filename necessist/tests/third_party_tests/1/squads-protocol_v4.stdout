41 candidates in 2 tests in 2 source files
$DIR/tests/suites/examples/create-mint.ts: dry running
$DIR/tests/suites/examples/create-mint.ts: mutilating
$DIR/tests/suites/examples/create-mint.ts: `await connection.confirmTransaction(airdropSig);` passed
$DIR/tests/suites/examples/create-mint.ts: `await connection.confirmTransaction(signature);` failed
$DIR/tests/suites/examples/create-mint.ts: `signature = await multisig.rpc.proposalCreate({
      connection,
      feePayer: members.voter,
      multisigPda,
      transactionIndex,
      creator: members.voter,
      programId,
    });` failed
$DIR/tests/suites/examples/create-mint.ts: `await connection.confirmTransaction(signature);` failed
$DIR/tests/suites/examples/create-mint.ts: `signature = await multisig.rpc.proposalApprove({
      connection,
      feePayer: members.voter,
      multisigPda,
      transactionIndex,
      member: members.voter,
      memo: "LGTM",
      programId,
    });` failed
$DIR/tests/suites/examples/create-mint.ts: `await connection.confirmTransaction(signature);` passed
$DIR/tests/suites/examples/create-mint.ts: `signature = await multisig.rpc.proposalApprove({
      connection,
      feePayer: members.almighty,
      multisigPda,
      transactionIndex,
      member: members.almighty,
      memo: "LGTM too",
      programId,
    });` failed
$DIR/tests/suites/examples/create-mint.ts: `await connection.confirmTransaction(signature);` failed
$DIR/tests/suites/examples/create-mint.ts: `signature = await multisig.rpc.vaultTransactionExecute({
      connection,
      feePayer: members.executor,
      multisigPda,
      transactionIndex,
      member: members.executor.publicKey,
      signers: [members.executor],
      sendOptions: { skipPreflight: true },
      programId,
    });` failed
$DIR/tests/suites/examples/create-mint.ts: `await connection.confirmTransaction(signature);` failed
$DIR/tests/suites/examples/create-mint.ts: `.fromAccountAddress(
      connection,
      multisigPda
    )` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.utils.toBigInt(multisigAccount.transactionIndex)` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.getTransactionPda({
      multisigPda,
      index: transactionIndex,
      programId,
    })` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.getVaultPda({
      multisigPda,
      index: 0,
      programId,
    })` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.requestAirdrop(
      vaultPda,
      lamportsForMintRent
    )` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.confirmTransaction(airdropSig)` passed
$DIR/tests/suites/examples/create-mint.ts: `.getEphemeralSignerPda({
      transactionPda,
      ephemeralSignerIndex: 0,
      programId,
    })` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.getLatestBlockhash()` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.createAccount({
          fromPubkey: vaultPda,
          newAccountPubkey: mintPda,
          space: MINT_SIZE,
          lamports: lamportsForMintRent,
          programId: TOKEN_2022_PROGRAM_ID,
        })` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.rpc.vaultTransactionCreate({
      connection,
      feePayer: members.proposer,
      multisigPda,
      transactionIndex,
      creator: members.proposer.publicKey,
      vaultIndex: 0,
      ephemeralSigners: 1,
      transactionMessage: testTransactionMessage,
      memo: "Create new mint",
      programId,
    })` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.confirmTransaction(signature)` failed
$DIR/tests/suites/examples/create-mint.ts: `.rpc.proposalCreate({
      connection,
      feePayer: members.voter,
      multisigPda,
      transactionIndex,
      creator: members.voter,
      programId,
    })` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.confirmTransaction(signature)` failed
$DIR/tests/suites/examples/create-mint.ts: `.rpc.proposalApprove({
      connection,
      feePayer: members.voter,
      multisigPda,
      transactionIndex,
      member: members.voter,
      memo: "LGTM",
      programId,
    })` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.confirmTransaction(signature)` passed
$DIR/tests/suites/examples/create-mint.ts: `.rpc.proposalApprove({
      connection,
      feePayer: members.almighty,
      multisigPda,
      transactionIndex,
      member: members.almighty,
      memo: "LGTM too",
      programId,
    })` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.confirmTransaction(signature)` failed
$DIR/tests/suites/examples/create-mint.ts: `.rpc.vaultTransactionExecute({
      connection,
      feePayer: members.executor,
      multisigPda,
      transactionIndex,
      member: members.executor.publicKey,
      signers: [members.executor],
      sendOptions: { skipPreflight: true },
      programId,
    })` nonbuildable
$DIR/tests/suites/examples/create-mint.ts: `.confirmTransaction(signature)` failed
$DIR/tests/suites/examples/immediate-execution.ts: dry running
$DIR/tests/suites/examples/immediate-execution.ts: mutilating
$DIR/tests/suites/examples/immediate-execution.ts: `tx.sign([members.almighty]);` failed
$DIR/tests/suites/examples/immediate-execution.ts: `await connection.confirmTransaction(signature);` failed
$DIR/tests/suites/examples/immediate-execution.ts: `.instructions.configTransactionCreate({
      multisigPda,
      transactionIndex,
      creator: members.almighty.publicKey,
      // Change threshold to 2.
      actions: [{ __kind: "ChangeThreshold", newThreshold: 2 }],
      programId,
    })` nonbuildable
$DIR/tests/suites/examples/immediate-execution.ts: `.instructions.proposalCreate({
      multisigPda,
      transactionIndex,
      creator: members.almighty.publicKey,
      programId,
    })` nonbuildable
$DIR/tests/suites/examples/immediate-execution.ts: `.instructions.proposalApprove({
      multisigPda,
      transactionIndex,
      member: members.almighty.publicKey,
      programId,
    })` nonbuildable
$DIR/tests/suites/examples/immediate-execution.ts: `.instructions.configTransactionExecute(
      {
        multisigPda,
        transactionIndex,
        member: members.almighty.publicKey,
        rentPayer: members.almighty.publicKey,
        programId,
      }
    )` nonbuildable
$DIR/tests/suites/examples/immediate-execution.ts: `.getLatestBlockhash()` nonbuildable
$DIR/tests/suites/examples/immediate-execution.ts: `.compileToV0Message()` nonbuildable
$DIR/tests/suites/examples/immediate-execution.ts: `.sign([members.almighty])` failed
$DIR/tests/suites/examples/immediate-execution.ts: `.sendTransaction(tx, {
      skipPreflight: true,
    })` nonbuildable
$DIR/tests/suites/examples/immediate-execution.ts: `.confirmTransaction(signature)` failed
$DIR/tests/suites/examples/immediate-execution.ts: `.fromAccountAddress(
      connection,
      multisigPda
    )` nonbuildable
